src/pipewire/pipewire.cc

```diff
@@ -159,36 +161,61 @@ void PipeWireOutput::pause(bool pause)
 {
     pw_thread_loop_lock(m_loop);
     pw_stream_set_active(m_stream, !pause);
+    clock_gettime(CLOCK_REALTIME, &m_time);
     pw_thread_loop_unlock(m_loop);
 }
 
 int PipeWireOutput::get_delay()
 {
-    return (m_buffer_at / m_stride + m_frames) * 1000 / m_rate;
+    int buff_time = (((m_buffer.len() + m_buffer_size) / m_stride) * 1000) / m_rate;
+
+    // Get time difference from last buffer fill
+    timespec time_now {};
+    clock_gettime(CLOCK_REALTIME, &time_now);
+
+    int time_diff =
+        (((time_now.tv_sec - m_time.tv_sec) * 1000000000) +
+         (time_now.tv_nsec - m_time.tv_nsec)) / 1000000;
+
+    if (time_diff < 0)
```

Comment:
`aud::clamp` would be convenient for the range-limiting.

Reply:
Thanks for advice, corrected in next version.